types/address.go

```diff
@@ -72,6 +72,23 @@ func (a *Address) Decode(decoder scale.Decoder) error {
 		return fmt.Errorf("decoding of Address with 0xfe prefix not supported")
 	}
 
+	if b != 0xfd && b != 0xfc {
```

Comment:
Main Address decoder logic.

Reply:
Where do you handle the AccountID case here? Address is an enum, where the variant is encoded in the first byte. Chains without the indices pallet don't use that enum at all, meaning you can't deduct anything from the first byte (the first byte is the little endian first byte of the public key, which could be `0xfd` without meaning it's an account index.

Reply:
I feel you need to use opts here as well.

Reply:
I need more clarity on this. Taking it offline.